src/legacy.rs

```diff
@@ -2,42 +2,51 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use crate::{
-    aesgcm::{AesGcmEceWebPush, AesGcmEncryptedBlock},
-    common::WebPushParams,
-    crypto::EcKeyComponents,
-    error::*,
-};
+pub use crate::aesgcm::AesGcmEncryptedBlock;
+use crate::{aesgcm::AesGcmEceWebPush, common::WebPushParams, crypto::EcKeyComponents, error::*};
 
 /// Encrypt a block using legacy AESGCM encoding.
 ///
-/// * `remote_pub` : the remote public key
-/// * `remote_auth` : the remote authorization token
-/// * `salt` : the locally generated random salt
+/// * `remote_pub` : The public key of the remote message recipient
+/// * `remote_auth` : The authentication secret of the remote message recipient
 /// * `data` : the data to encrypt
 ///
+/// You should only use this function if you know that you definitely need
+/// to use the legacy format. The [`encrypt`](crate::encrypt) function should
+/// be preferred where possible.
+///
 pub fn encrypt_aesgcm(
     remote_pub: &[u8],
     remote_auth: &[u8],
-    salt: &[u8],
     data: &[u8],
 ) -> Result<AesGcmEncryptedBlock> {
     let cryptographer = crate::crypto::holder::get_cryptographer();
     let remote_key = cryptographer.import_public_key(remote_pub)?;
     let local_key_pair = cryptographer.generate_ephemeral_keypair()?;
+    // Randomly select a padding length, without exceeding the record size.
+    // The magic constants here will go away in a future refactor.
     let mut padr = [0u8; 2];
     cryptographer.random_bytes(&mut padr)?;
-    // since it's a sampled random, endian doesn't really matter.
-    let pad = ((usize::from(padr[0]) + (usize::from(padr[1]) << 8)) % 4095) + 1;
-    let params = WebPushParams::new(4096, pad, Vec::from(salt));
+    let mut pad_length = ((usize::from(padr[0]) + (usize::from(padr[1]) << 8)) % 4095) + 1;
```

Comment:
nit: We do this here and in [`lib.rs`](https://github.com/mozilla/rust-ece/pull/52/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R52). Would it be worth making this into a `pub(crate)` function?

Reply:
I intend to propose a change to how this padding works in a follow-up PR, but yeah, it's probably worth pulling out a quick helper in the meantime.